sa/database_test.go

```diff
@@ -5,35 +5,75 @@ import (
 	"errors"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/letsencrypt/boulder/test"
 	"github.com/letsencrypt/boulder/test/vars"
 )
 
 func TestInvalidDSN(t *testing.T) {
-	_, err := NewDbMap("invalid", 0)
+	_, err := NewDbMap("invalid", DbSettings{})
 	test.AssertError(t, err, "DB connect string missing the slash separating the database name")
 }
 
 var errExpected = errors.New("expected")
 
-func TestMaxOpenConns(t *testing.T) {
+func TestDbSettings(t *testing.T) {
 	oldSetMaxOpenConns := setMaxOpenConns
+	oldSetMaxIdleConns := setMaxIdleConns
```

Comment:
This kind of code duplication really suggests to me that what we actually need is a full `db.mockWrappedMap`. I don't think that should be in-scope for this change, but maybe file a bug for it and add a TODO pointing to that bug.